StRoot/St_geant_Maker/St_geant_Maker.cxx

```diff
@@ -1177,6 +1180,17 @@ int St_geant_Maker::InitRun(int run){
 }
 //_____________________________________________________________________________
 int St_geant_Maker::Make() {
+
+  static bool first = true;
+  if ( first ) {    // Event generator may have been plugged into chain after Init()
+
+    auto* stack = GetDataSet("PrimaryMakerStack");
+    auto* data  = GetDataSet("PDG_Database");
+    if ( stack != 0 )       AgStarReader::Instance().SetStack( (StarParticleStack*)stack->GetObject() );
+    if ( data  != 0 )       AgStarReader::Instance().SetParticleData( (StarParticleData*)data->GetObject() );
```

Comment:
This is one of those cases where it is hard to avoid.  If I use static / dynamic casts, then the dynamic linker complains that StarParticleStack / Data are not completely defined when I load the St_geant_Maker library.  Same result if I push the cast down into the AgStarReader class.  The crime against good coding style is not the use of the c-style cast, but rather the use of the singleton pattern for AgStarReader.  Would prefer to keep the c-style cast here, rather than re-engineer the fortran interface.

Reply:
Are you trying to avoid `#include "StarGenerator/UTIL/StarParticleData.h"` and `#include "StarGenerator/BASE/StarParticleStack.h"` from `StRoot/St_geant_Maker/St_geant_Maker.cxx`?

Reply:
Yes.  Also trying to avoid any run time dependence between the two libraries, so that
St_geant_Maker can be loaded without StarGenerator, and StarGenerator can be loaded w/out
loading St_geant_Maker.

(Also, not sure if replying in email gets logged as part of the discussion on github?  If so, this may be redundant.)